packages/app/src/providers/utils.ts

```diff
@@ -8,13 +8,14 @@ import type {
 import { DTypeClass } from '@h5web/shared/hdf5-models';
 import type { OnProgress } from '@h5web/shared/react-suspense-fetch';
 import type { AxiosProgressEvent } from 'axios';
+import { isAxiosError } from 'axios';
 
 import type { DataProviderApi } from './api';
 
 export const CANCELLED_ERROR_MSG = 'Request cancelled';
 
 export function typedArrayFromDType(dtype: DType) {
-  if (isEnumType(dtype)) {
+  if (isEnumType(dtype) || isBoolType(dtype)) {
```

Comment:
Sorry for the dumb question: isn't bool a particular type of `enum` ? So that `isEnumType` should encompass bool ?

Reply:
We split the two in our internal types and the purpose of these guards is to narrow our internal types. We need two separate types so `Primitive<BooleanType>` can resolve to `number | boolean`. But we may be able to reconsider this choice with https://github.com/silx-kit/h5web/issues/1679#issuecomment-2333937430